src/services/components/index.ts

```diff
@@ -32,27 +32,76 @@ const HYPERVIEW_COMPONENTS = [
   HvWebView,
 ];
 
-const reducer = (registry: ComponentRegistry, component: HvComponent) => ({
-  ...registry,
-  [component.namespaceURI]: {
-    ...registry[component.namespaceURI],
-    ...ComponentsInternal.registerComponent(component),
-  },
-});
-
-export const getRegistry = (
-  components: HvComponent[] = [],
-): ComponentRegistry =>
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [...HYPERVIEW_COMPONENTS, ...components].reduce<Record<string, any>>(
-    reducer,
-    {},
-  );
-
-export const getFormRegistry = (
-  components: HvComponent[] = [],
-): ComponentRegistry =>
-  [...HYPERVIEW_COMPONENTS, ...components]
-    .filter(c => Object.prototype.hasOwnProperty.call(c, 'getFormInputValues'))
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    .reduce<Record<string, any>>(reducer, {});
+export class ComponentRegistry {
```

Comment:
nit: since the module name already convey the namespace, it might simplify usage throughout to rename this class to `Registry`.